webpack.common.js

```diff
@@ -66,6 +87,13 @@ module.exports = {
     plugins: [
         new HTMLWebpackPlugin({ template: path.resolve(__dirname, 'src/index.html') }),
         new FaviconsWebpackPlugin(path.resolve(__dirname, 'src/favicon.png')),
+        new webpack.EnvironmentPlugin({
+            GraphHopperApiKey: apikeys.graphhopper, // use no default to throw exception if missing
```

Comment:
What about using the web app with graphhopper running on localhost? Maybe we should already rename `apikeys.js` to `config.js` and include host and port and rather make the graphhopper key optional?

Reply:
Sure we can make these configurable too and rename the file name.

But I'm unsure if we should make the key optional in this UI-only repo as we would then have to explain people on how to set up the graphhopper routing server. Instead I would require the API key and link to graphhopper routing server installation instructions if this is not wanted.

Reply:
And I'm not saying this from some suspicious commercial interests ;D ... I mean we should also require MapTiler API key (like deck.gl requires mapbox api keys), because we expect no one to setup OpenMapTiles only to run this UI project.

Reply:
Ok and when we want to use it with local GraphHopper we just need to enter a dummy key?

Reply:
I wonder who is willing to get an api key just to develop the UI, I'd rather use a local GraphHopper instance and instructions how to set this up can be easily found at the GH repo. And btw don't you want to move the web app into the GH repo anyway? I mean once this replaces the old web app it would be nice to have it out of the box when setting up a GH server?!

Reply:
> Ok and when we want to use it with local GraphHopper we just need to enter a dummy key?

Yes, or we detect the host and if it contains localhost we ignore this requirement?

> I'd rather use a local GraphHopper instance and instructions how to set this up can be easily found at the GH repo

IMO if you are a front end dev you don't want to mess with a JVM ;) ... also again, why require a maptiler key and not the GraphHopper key?

> And btw don't you want to move the web app into the GH repo anyway?

Not so sure anymore - what do you think? Having separate repos has the advantage of a different release schedule. And in GH core we could fetch a working release. Of course has also disadvantages ...

> I mean once this replaces the old web app it would be nice to have it out of the box when setting up a GH server?!

Of course, sure.

Reply:
As long as it is somehow possible to develop against localhost I'm happy :) Now that we have a config file I'd also wish we can configure the host in this config file, but if not we can still modify `Api.ts`.

> also again, why require a maptiler key and not the GraphHopper key?

I guess this is a frontend to GraphHopper so some interest/knowledge about GraphHopper can be expected I think. Also one can develop using openstreetmap tiles without a key.

> Not so sure anymore - what do you think? Having separate repos has the advantage of a different release schedule. And in GH core we could fetch a working release. Of course has also disadvantages ...

I'm also not sure, but I don't see much reason for change compared to what we are doing now (keep the webapp inside the graphhopper repo). The main requirement is that the webapp is running once we start the GH server, but you are right we can probably fetch a working release for this.